stack_data/core.py

```diff
@@ -709,6 +761,14 @@ def lines(self) -> List[Union[Line, LineGap]]:
                     and pieces[1] != self.scope_pieces[1]
             ):
                 result.append(LINE_GAP)
+            elif (prev_piece and add_empty_lines and piece.start > prev_piece.stop):
+                lines = []  # type: List[Union[EmptyLine, BlankLineRange]]
+                if self.blank_lines == BlankLines.LINE_NUMBER:
+                    lines.append(BlankLineRange(prev_piece.stop, piece.start-1))
+                else:  # BlankLines.VISIBLE
+                    for lineno in range(prev_piece.stop, piece.start):
+                        lines.append(EmptyLine(self, lineno))
+                result.extend(lines)
```

Comment:
```suggestion
            elif prev_piece and add_empty_lines and piece.start > prev_piece.stop:
                if self.blank_lines == BlankLines.LINE_NUMBER:
                    result.append(BlankLineRange(prev_piece.stop, piece.start-1))
                else:  # BlankLines.VISIBLE
                    for lineno in range(prev_piece.stop, piece.start):
                        result.append(EmptyLine(self, lineno))
```

`prev_piece` could be replaced by `pieces[i-1]` if you check that `i > 0`. But it probably wouldn't be an improvement.